PWGHF/TableProducer/candidateCreatorXicToXiPiPi.cxx

```diff
@@ -0,0 +1,655 @@
+      // create cascade track
+      o2::track::TrackParCov trackCasc;
+      if (trackPionFromXi.sign() > 0) {
+        trackCasc = o2::track::TrackParCov(vertexCasc, pVecCasc, covCasc, 1, true);
+      } else if (trackPionFromXi.sign() < 0) {
+        trackCasc = o2::track::TrackParCov(vertexCasc, pVecCasc, covCasc, -1, true);
+      } else {
+        continue;
+      }
+      trackCasc.setAbsCharge(1);
+      trackCasc.setPID(o2::track::PID::XiMinus);
+
+      //----------------------------fit SV and create XicPlus track------------------
+      auto trackParCovCharmBachelor0 = getTrackParCov(trackCharmBachelor0);
+      auto trackParCovCharmBachelor1 = getTrackParCov(trackCharmBachelor1);
+
+      // reconstruct the 3-prong secondary vertex
+      if (df.process(trackCasc, trackParCovCharmBachelor0, trackParCovCharmBachelor1) == 0) {
+        continue;
+      }
+
+      //----------------------------calculate physical properties-----------------------
+      // Charge of charm baryon
+      int signXic = casc.sign() < 0 ? +1 : -1;
+
+      // get SV properties
+      const auto& secondaryVertex = df.getPCACandidate();
+      auto chi2SV = df.getChi2AtPCACandidate();
+      auto covMatrixSV = df.calcPCACovMatrixFlat();
+
+      // get track momenta
+      trackParCovCharmBachelor0 = df.getTrack(1);
+      trackParCovCharmBachelor1 = df.getTrack(2);
+      trackCasc = df.getTrack(2);
+      std::array<float, 3> pVecXi;
+      std::array<float, 3> pVecPi0;
+      std::array<float, 3> pVecPi1;
+      trackCasc.getPxPyPzGlo(pVecXi);
+      trackParCovCharmBachelor0.getPxPyPzGlo(pVecPi0);
+      trackParCovCharmBachelor1.getPxPyPzGlo(pVecPi1);
+
+      // get invariant mass of Xic candidate
+      auto arrayMomenta = std::array{pVecXi, pVecPi0, pVecPi1};
+      massXiPiPi = RecoDecay::m(std::move(arrayMomenta), std::array{massXiMinusFromPdg, massPionFromPdg, massPionFromPdg});
+
+      // get track impact parameters
+      // This modifies track momenta!
+      auto primaryVertex = getPrimaryVertex(collision);
+      auto covMatrixPV = primaryVertex.getCov();
+      // calculate impact parameter
+      o2::dataformats::DCA impactParameterCasc;
+      o2::dataformats::DCA impactParameter0;
+      o2::dataformats::DCA impactParameter1;
+      trackCasc.propagateToDCA(primaryVertex, bz, &impactParameterCasc);
+      trackParCovCharmBachelor0.propagateToDCA(primaryVertex, bz, &impactParameter0);
+      trackParCovCharmBachelor1.propagateToDCA(primaryVertex, bz, &impactParameter1);
+
+      // calculate cosine of pointing angle
+      std::array<float, 3> pvCoord = {collision.posX(), collision.posY(), collision.posZ()};
+      double cpaLambda = casc.v0cosPA(collision.posX(), collision.posY(), collision.posZ());
+      double cpaXYLambda = RecoDecay::cpaXY(pvCoord, vertexV0, pVecV0);
+      double cpaXi = casc.casccosPA(collision.posX(), collision.posY(), collision.posZ());
+      double cpaXYXi = RecoDecay::cpaXY(pvCoord, vertexCasc, pVecCasc);
```

Comment:
not a strong opinion for this, just to tell you that `RecoDecay` implements also a `cpa` function (see here: https://github.com/AliceO2Group/O2Physics/blob/b06b9c01a6df7386eff1b0eb0a98ac643464996b/Common/Core/RecoDecay.h#L223C1-L241C4)

Reply:
Yes. These are dynamic columns that use the RecoDecay::cpa functions. I chose the dynamic columns out of convenience, as I don't have to pass as many arguments  in this case, the rest of them are already handled.